src/smartpeak/include/SmartPeak/core/Parameters.h

```diff
@@ -184,6 +185,12 @@ namespace SmartPeak
       @brief return default value of the parameter (if schema is assigned).
     */
     const std::string getDefaultValueAsString() const { return schema_ ? schema_->getValueAsString() : getValueAsString(); };
+    
+    /**
+      @brief return current parameter values in the same order as the columns : headers = { "name", "type", "value", "restrictions" }
+    */
+    const std::vector<std::string> getAllValues() const {
+      return {this->getName(), this->getType(), this->getValueAsString(), this->getRestrictionsAsString()}; };
```

Comment:
I don't like too much introducing a method there to fit a UI constraint (column order). It should be up to the UI to order that field on their constraint.

Reply:
a much more expensive operation and cluttery when we push back the values in the draw method every time we need these values.

the constraint is based on the static const vector you've introduced : `static const std::vector<std::string> headers = { "function", "name", "type", "value", "restrictions" };`

Reply:
This headers vector is at the right place, on UI side, in ParametersTableWidget, as it sticks with the UI presentation, defining an order in the columns. i feel getAllValue should be there as well. what about making private method of ParametersTableWidget getAllValue taking a Parameter as an argument?

Reply:
i don't see the need of having a constant object getting recreated every time a frame is drawn, if it's existence is not a necessity for modifying it then we'd better move it out of the draw() function especially with nested loops. getAllValues_() has been moved to ParametersTableWidget.